Let code owners merge PRs

Eirenliel opened the pull request:
Adds CODEOWNERS file and assigns some codeownders. The codeowners can merge PRs by posting `LGTM` if all checks passed using https://github.com/marketplace/actions/code-owner-self-merge?version=1.6.5

Eirenliel commented:
suggestions for code owners are welcomed

Louka3000 commented:
I think @loucass003 should be code owner for server config since he wrote all of it: `server/src/main/java/dev/slimevr/config`

and I'd like to be code owner on skeleton stuff: `server/src/main/java/dev/slimevr/vr/processor` (which is changing to `server/src/main/java/dev/slimevr/tracking/processor` with my refactor...) and tracker filtering: `server/src/main/java/dev/slimevr/filtering`

Eirenliel commented:
👍

TheButlah commented on .github/CODEOWNERS:
```diff
@@ -0,0 +1,23 @@
+/server/src/main/java/dev/slimevr/vr/processor/ @Louka3000
+/server/src/main/java/dev/slimevr/filtering/ @Louka3000
+
+server/src/main/java/dev/slimevr/config/ @loucass003
```
```suggestion
server/src/main/java/dev/slimevr/config/ @loucass003

```

TheButlah commented on .github/workflows/codeowners-merge.yml:
```diff
@@ -0,0 +1,18 @@
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          merge_method: squash
+          if_no_maintainers_assign: '@Eirenliel @ButterscotchV'
```
```suggestion
          if_no_maintainers_assign: '@Eirenliel @ButterscotchV'
```

ImUrX commented on .github/CODEOWNERS:
```diff
@@ -0,0 +1,23 @@
+
+# Make Loucas the owner of all GUI stuff
+/gui/ @loucass003
+
```
```suggestion

# Rust part of the GUI
/gui/src-tauri/ @ImUrX @TheButlah 

```

Louka3000 commented:
_Actually, why does Butter get full merge perms on all of /server, shouldn’t I also do… I even wrote more code :nya_a:_

Louka3000 commented:
Important stuff from Discord:

With the current way it is, I won’t be able to do anything except hotfixes to these specific packages, since implementing anything new would require config or protocol/rpc changes. 

So, for example, even if 99% of my code in a PR to add a new skeleton config toggle is in the skeleton package, if it has a line of code in RPCSettingsHandler to make it work with the protocol, it’d absolutely need Butter or Eiren to merge…

Something like https://github.com/marketplace/actions/auto-request-review would allow to automatically request reviews when a PR touches a certain part of the code.
I propose Butter and I have merge perms for server, and moving the specific perms (Autobone, BVH, Skeleton, OSC and Filtering) to auto-reviews.

Btw, having perms to merge `gui/src/components/settings` would be nice too, since I’ve been touching that part a lot

Louka3000 reviewed (commented):
^

Eirenliel commented:
> The problem thus is that my codeowner perms are useless except for small hotfixes inside given packages, as any change that brings new stuff to the server will need code change outside of those packages.

We were talking about letting you to merge language changes in the first place. This system is a workaround from giving permissions to merge on everything. I'm trying to make a "well you can sort at least some of it without me and i'll still sleep well" kind of system.

Louka3000 commented:
My original proposal was just to be able to merge on x approving reviews.
Here, you’re giving Butter ownership of all server code, which means we could already merge full-fledged features or substantial code changes, but why only Butter? Feels unfair to me considering most recent server code is mine…

Eirenliel commented:
> With the current way it is, I won’t be able to do anything except hotfixes to these specific packages, since implementing anything new would require config or protocol/rpc changes.

You shouldn't review and merge your own PRs either way. This is for when someone else changes your code, you can be an authority on merging it. That said, even then I want a policy that thrid eyes should look at it first.

Eirenliel commented:
I added a bunch of owners and rules and removed the system with LGTM, instead @TheButlah, @Louka3000, I and @ButterscotchV are now maintainers and can merge if branch protection rules are satisfied. Which I changed to require 2 reviews and code owner reviews.

Louka3000 commented:
From Discord:
- All code owners for the PR must approve (forced by GH)
- Need at least 2 total approving reviews by anyone in the GH contributor team (forced by GH)
- Need checks to pass and no change requests (forced by GH)
- No change request for a day (not forced by GH)
